Possible semver breaking change in serde 1.0.157

When pinning the version of `serde` in Cargo.toml to `1.0.156`, my crate compiles, switching to `1.0.157` results in an error in my `influxdb2` dependency:
```bash
error: unexpected end of input, unexpected token in nested attribute, expected ident
  --> /home/midas/.cargo/registry/src/github.com-1ecc6299db9ec823/influxdb2-0.3.5/src/models/ast/dialect.rs:16:13
   |
16 |     #[serde()]
   |             ^

   Compiling opentelemetry-otlp v0.11.0
error[E0277]: the trait bound `Dialect: buckets::_::_serde::Serialize` is not satisfied
    --> /home/midas/.cargo/registry/src/github.com-1ecc6299db9ec823/influxdb2-0.3.5/src/models/query.rs:10:35
     |
10   | #[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
     |                                   ^^^^^^^^^ the trait `buckets::_::_serde::Serialize` is not implemented for `Dialect`
...
20   |     /// Dialect
     |     ----------- required by a bound introduced by this call
     |
     = help: the following other types implement trait `buckets::_::_serde::Serialize`:
               &'a T
               &'a mut T
               ()
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
               (T0, T1, T2, T3, T4)
               (T0, T1, T2, T3, T4, T5)
             and 208 others
     = note: required for `std::option::Option<Dialect>` to implement `buckets::_::_serde::Serialize`
note: required by a bound in `buckets::_::_serde::ser::SerializeStruct::serialize_field`
    --> /home/midas/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.157/src/ser/mod.rs:1901:12
     |
1901 |         T: Serialize;
     |            ^^^^^^^^^ required by this bound in `buckets::_::_serde::ser::SerializeStruct::serialize_field`

error[E0277]: the trait bound `Dialect: buckets::_::_serde::Deserialize<'_>` is not satisfied
    --> /home/midas/.cargo/registry/src/github.com-1ecc6299db9ec823/influxdb2-0.3.5/src/models/query.rs:22:18
     |
22   |     pub dialect: Option<crate::models::ast::Dialect>,
     |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `buckets::_::_serde::Deserialize<'_>` is not implemented for `Dialect`
     |
     = help: the following other types implement trait `buckets::_::_serde::Deserialize<'de>`:
               &'a Path
               &'a [u8]
               &'a serde_json::value::RawValue
               &'a str
               ()
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
             and 219 others
     = note: required for `std::option::Option<Dialect>` to implement `buckets::_::_serde::Deserialize<'_>`
note: required by a bound in `next_element`
    --> /home/midas/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.157/src/de/mod.rs:1729:12
     |
1729 |         T: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `next_element`

error[E0277]: the trait bound `Dialect: buckets::_::_serde::Deserialize<'_>` is not satisfied
    --> /home/midas/.cargo/registry/src/github.com-1ecc6299db9ec823/influxdb2-0.3.5/src/models/query.rs:22:18
     |
22   |     pub dialect: Option<crate::models::ast::Dialect>,
     |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `buckets::_::_serde::Deserialize<'_>` is not implemented for `Dialect`
     |
     = help: the following other types implement trait `buckets::_::_serde::Deserialize<'de>`:
               &'a Path
               &'a [u8]
               &'a serde_json::value::RawValue
               &'a str
               ()
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
             and 219 others
     = note: required for `std::option::Option<Dialect>` to implement `buckets::_::_serde::Deserialize<'_>`
note: required by a bound in `next_value`
    --> /home/midas/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.157/src/de/mod.rs:1868:12
     |
1868 |         V: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `next_value`

error[E0277]: the trait bound `Dialect: buckets::_::_serde::Deserialize<'_>` is not satisfied
  --> /home/midas/.cargo/registry/src/github.com-1ecc6299db9ec823/influxdb2-0.3.5/src/models/query.rs:20:5
   |
20 |     /// Dialect
   |     ^^^^^^^^^^^ the trait `buckets::_::_serde::Deserialize<'_>` is not implemented for `Dialect`
   |
   = help: the following other types implement trait `buckets::_::_serde::Deserialize<'de>`:
             &'a Path
             &'a [u8]
             &'a serde_json::value::RawValue
             &'a str
             ()
             (T0, T1)
             (T0, T1, T2)
             (T0, T1, T2, T3)
           and 219 others
   = note: required for `std::option::Option<Dialect>` to implement `buckets::_::_serde::Deserialize<'_>`
note: required by a bound in `buckets::_::_serde::__private::de::missing_field`
  --> /home/midas/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.157/src/private/de.rs:22:8
   |
22 |     V: Deserialize<'de>,
   |        ^^^^^^^^^^^^^^^^ required by this bound in `buckets::_::_serde::__private::de::missing_field`
```

Looking at the [changes](https://github.com/serde-rs/serde/compare/v1.0.156...v1.0.157), I don't really understand what's going on, so I'm not able to tell whether this is an unexpected issue coming from `serde` or if `influxdb2` was misusing `serde` before.
But from what I can tell it seems that putting an attribute inside a derive macro was allowed first, but not anymore, i.e.:
```
#[derive(Clone, Debug, PartialEq, Default, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Dialect {
    #[serde()]
    pub annotations: Vec<Annotations>,
}
```

--- a/serde_derive/src/internals/attr.rs
+++ b/serde_derive/src/internals/attr.rs
@@ -307,6 +307,12 @@ impl Container {
                 continue;
             }
 
+            if let syn::Meta::List(meta) = &attr.meta {
+                if meta.tokens.is_empty() {
+                    continue;
+                }
+            }
+
             if let Err(err) = attr.parse_nested_meta(|meta| {
                 if meta.path == RENAME {
                     // #[serde(rename = "foo")]
@@ -762,6 +768,12 @@ impl Variant {
                 continue;
             }
 
+            if let syn::Meta::List(meta) = &attr.meta {
+                if meta.tokens.is_empty() {
+                    continue;
+                }
+            }
+
             if let Err(err) = attr.parse_nested_meta(|meta| {
                 if meta.path == RENAME {
                     // #[serde(rename = "foo")]
@@ -1031,6 +1043,12 @@ impl Field {
         for attr in &field.attrs {
             if attr.path() != SERDE {
                 continue;
+            }
+
+            if let syn::Meta::List(meta) = &attr.meta {
+                if meta.tokens.is_empty() {
+                    continue;
+                }
             }
 
             if let Err(err) = attr.parse_nested_meta(|meta| {

--- a/test_suite/tests/regression/issue2415.rs
+++ b/test_suite/tests/regression/issue2415.rs
@@ -1,0 +1,5 @@
+use serde_derive::Serialize;
+
+#[derive(Serialize)]
+#[serde()]
+pub struct S;

